src/auth/index.ts

```
@@ -0,0 +1,144 @@
+    Username: req.body.username,
+    Password: req.body.password,
+  };
+  const authenticationDetails = new AuthenticationDetails(authenticationData);
+
+  // Create Cognito user object
+  const userData = {
+    Username: username,
+    Pool: userPool,
+  };
+
+  const cognitoUser = new CognitoUser(userData);
+
+  // Authenticate user
+
+  cognitoUser.authenticateUser(authenticationDetails, {
+    onSuccess: (session) => {
+      const options: jwt.SignOptions = {
+        header: { alg: 'HS256' },
+      };
+      const load = {
+        name: session.getIdToken().payload.email,
+        email: session.getIdToken().payload.email,
+        picture: null,
+        sub: session.getIdToken().payload.sub,
+        userId: session.getIdToken().payload.sub,
+        clientId: session.getIdToken().payload.aud,
+      };
+
+      const jwtToken = jwt.sign(load, key as string, options);
+      res.status(200).send(jwtToken);
+    },
+    onFailure: () => {
+      // Authentication failed
+      res.status(401).json({ error: 'Authentication failed' });
+    },
+    newPasswordRequired: () => {
+      // Set the new password for the user
+      cognitoUser.completeNewPasswordChallenge(newPassword, null, {
+        onSuccess: (session) => {
+          // Password updated successfully, generate a session token
+          const options: jwt.SignOptions = {
+            header: { alg: 'HS256' },
+          };
+          const load = {
+            name: session.getIdToken().payload.email,
+            email: session.getIdToken().payload.email,
+            picture: null,
+            sub: session.getIdToken().payload.sub,
+            userId: session.getIdToken().payload.sub,
+            clientId: session.getIdToken().payload.aud,
+          };
+
+          const jwtToken = jwt.sign(load, key as string, options);
+          res.status(200).send(jwtToken);
+        },
+        onFailure: () => {
+          res.status(500).json({ error: 'Error setting new password' });
```

Again lets not swallow the error. Make sure we use logger.error to log the errors and also tell the sender why it failed.